precompile/warp_messenger.go

```diff
@@ -0,0 +1,444 @@
+	SubmitMessageEventID = "da2b1cd3e6664863b4ad90f53a4e14fca9fc00f3f0e01e5c7b236a4355b6591a" // Keccack256("SubmitMessage(bytes32,uint256)")
+)
+
+// Reference imports to suppress errors from unused imports. This code and any unnecessary imports can be removed.
+var (
+	_ = errors.New
+	_ = big.NewInt
+)
+
+// Singleton StatefulPrecompiledContract and signatures.
+var (
+	_ StatefulPrecompileConfig = &WarpMessengerConfig{}
+
+	WarpMessengerABI        abi.ABI                     // will be initialized by init function
+	WarpMessengerPrecompile StatefulPrecompiledContract // will be initialized by init function
+
+	// Default stake threshold for aggregate signature verification. (67%)
+	WarpQuorumNumerator   = uint64(67)
+	WarpQuorumDenominator = uint64(100)
+
+	signedMessages []*warp.Message
+
+	ErrMissingStorageSlots       = errors.New("missing access list storage slots from precompile during execution")
+	ErrInvalidMessageIndex       = errors.New("invalid message index")
+	ErrInvalidSignature          = errors.New("invalid aggregate signature")
+	ErrMissingProposerVMBlockCtx = errors.New("missing proposer VM block context")
+	ErrWrongChainID              = errors.New("wrong chain id")
+	ErrInvalidQuorumDenominator  = errors.New("quorum denominator can not be zero")
+	ErrGreaterQuorumNumerator    = errors.New("quorum numerator can not be greater than quorum denominator")
+)
+
+// WarpMessengerConfig implements the StatefulPrecompileConfig
+// interface while adding in the WarpMessenger specific precompile address.
+type WarpMessengerConfig struct {
+	UpgradeableConfig
+	InitialWarpConfig *InitialWarpConfig `json:"initialWarpConfig,omitempty"`
+}
+
+// WarpMessage is an auto generated low-level Go binding around an user-defined struct.
+type WarpMessage struct {
+	OriginChainID       [32]byte
+	OriginSenderAddress [32]byte
+	DestinationChainID  [32]byte
+	DestinationAddress  [32]byte
+	Payload             []byte
+}
+
+type GetVerifiedWarpMessageOutput struct {
+	Message WarpMessage
+	Success bool
+}
+
+type SendWarpMessageInput struct {
+	DestinationChainID [32]byte
+	DestinationAddress [32]byte
+	Payload            []byte
+}
+
+type InitialWarpConfig struct {
+	QuorumNumerator   uint64 `json:"quorumNumerator,omitempty"`
+	QuorumDenominator uint64 `json:"quorumDenominator,omitempty"`
+}
+
+func (i *InitialWarpConfig) Verify() error {
+	switch {
+	case i.QuorumDenominator == uint64(0):
+		return ErrInvalidQuorumDenominator
+	case i.QuorumNumerator > i.QuorumDenominator:
+		return ErrGreaterQuorumNumerator
+	default:
+		return nil
+	}
+}
+
+func (i *InitialWarpConfig) Equal(other *InitialWarpConfig) bool {
+	if other == nil {
+		return false
+	}
+
+	return i.QuorumNumerator == other.QuorumNumerator && i.QuorumDenominator == other.QuorumDenominator
+}
+
+func (i *InitialWarpConfig) Configure(state StateDB) {
+	WarpQuorumNumerator = i.QuorumNumerator
+	WarpQuorumDenominator = i.QuorumDenominator
+}
```

Comment:
This is incorrect. We cannot just set global variables when `Configure` is called. `Configure` is called while verifying a block, which is before it gets accepted and there may be multiple different blocks in processing that could trigger `Configure` getting called.

If we need to use this in the future, we should store it in the state in the same way that we do for the other precompiles when they save config information.